### DISCUSSION TITLE
Using picocli to capture undefined command with arguments

### DISCUSSION THREAD

**Discussion by Matiako** [remkop/picocli#2124] [Q&A] [answered]
Hello, 

can I use picocli to capture any (undefined) command and pass it further down the line with arguments attached to it?
I would like to use picocli to parse command line and without matching it to any specific command  take string at index 0 and save it as a command  in my request and take string with index 2,3,4,5 etc. and attach them as an arguments to request .

**Comment by remkop** [comment, marked as answer]
@Matiako I don't think picocli can provide any value in the use case you describe.
Picocli's value is to parse the predefined options for predefined commands and provide help to end users explaining which options are available for which command.

From your description, all you need is to pass the user input as a String array to your downstream component...

**Reply by Matiako** [reply]
Thank You very much for fast response.
